pkg/nsx/services/subnet/subnet.go

```diff
@@ -155,15 +155,17 @@ func (service *SubnetService) createOrUpdateSubnet(obj client.Object, nsxSubnet
 	// Failure of CheckRealizeState may result in the creation of an existing Subnet.
 	// For Subnets, it's important to reuse the already created NSXSubnet.
 	// For SubnetSets, since the ID includes a random value, the created NSX Subnet needs to be deleted and recreated.
-	if err = realizeService.CheckRealizeState(backoff, *nsxSubnet.Path, "RealizedLogicalSwitch"); err != nil {
-		log.Error(err, "Failed to check subnet realization state", "ID", *nsxSubnet.Id)
-		// Delete the subnet if realization check fails, avoiding creating duplicate subnets continuously.
-		deleteErr := service.DeleteSubnet(*nsxSubnet)
-		if deleteErr != nil {
-			log.Error(deleteErr, "Failed to delete Subnet after realization check failure", "ID", *nsxSubnet.Id)
-			return "", fmt.Errorf("realization check failed: %v; deletion failed: %v", err, deleteErr)
+	for _, entity := range realizestate.RealizedSubnetEntities {
+		if err = realizeService.CheckRealizeState(backoff, *nsxSubnet.Path, entity); err != nil {
```

Comment:
Can we update CheckRealizeState to accept entity array, so that we do not need to call the API for 3 times while the return actually contains all the information we need?